### PR TITLE
Refactor usb cam library with no ros deps

### PR DESCRIPTION
**Pull request by flynneva**
- Refactor `usb_cam` library to be  bit more understandable
- Add some basic unit tests
- Test to see if we can remove all ROS deps from usb_cam lib and only use ROS in the node wrappers

### CODE REVIEW

**Comment by flynneva** [comment]
@twdragon I refactored the code just a bit to be a bit more readable and added the beginning of some unit tests that we can add onto later.

The `usb_cam` library is already pretty well separated actually from the ROS code and from what I can tell only uses ROS-specific code for the timestamp and the logger. These should be pretty easy to get rid of I think but am interested in your feedback here on the best way to do that.

Should we try and template the code? So we could specify a ROS-clock type to use in `usb_cam`? Or should we try and return the time in some other way (e.g. `std::chrono::time_point`) and provide converter functions for each ROS-specific version? The later might be easier to implement, and also give us the benefit of being able to use the `usb_cam` library without ROS at all which should make integration tests a bit easier.

**Comment by twdragon** [comment]
@flynneva great work! Thank you a lot! Please give me some time to examine the code and adapt it, then it would be also a great unified release!

**Comment by flynneva** [comment]
@twdragon I added some more tests and I _think_ the `usb_cam.hpp` and `usb_cam.cpp` now have _no ROS deps_...so this means we could reuse this library for both ROS 1 and ROS 2 if needed / not have to change it if ROS 2 changes some API in the future.

To make this happen I had to remove the use of ROS 2 `rclcpp::Clock` and the msgs needed to store the timestamps of each frame.  I checked the new implementation and it seems to work...but let me know what you think / if you see anything that could be improved.

One thing I still am unsure about is how to "simulate" a `/dev/ttyUSB0` camera for integration tests. I'll look into this over the next week to see if there are already testing strategies out there that we could use.

**Comment by twdragon** [comment]
@flynneva great! I will start in some days, thank you for this contribution! I hope to make things work.

**Review by twdragon — APPROVED** [review]
I see it makes the code much mere readable. Thank you, and I will do my best to adapt it

**Comment by flynneva** [comment]
@twdragon I squeezed  one more change into this PR by adding a simple integration test to round out the first layer of tests we add here. This test can be improved later on once we are able to simulate v4l2 devices on the CI runners.

File: src/usb_cam.cpp [hunk 1]
```diff
@@ -26,48 +26,48 @@
 // ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
 
+#define CLEAR(x) memset(&(x), 0, sizeof(x))
```
Comment (twdragon) [inline src/usb_cam.cpp:29]: Could we consider a template instead?

**Review by twdragon — APPROVED** [review]
@flynneva great work again! I see an excellent opportunity to recover portability for ROS 1 node too and will have a look very soon! Now I am ready to approve this PR

File: src/usb_cam.cpp (hunk 1 above)
Comment (flynneva) [reply src/usb_cam.cpp:29]: @twdragon yes for sure! This was just the existing code as before but now after the refactor we can improve it

**Pull request merged**